### ISSUE TITLE
Use AJAX for Console Rule

### ISSUE THREAD

**Issue by BrainBuzzer** [fossasia/susi_server#44]
The idea is to use AJAX with console rule app so that all things would be made asynchronous. Currently, it redirects to other links and does not do everything on same page.

**Comment by AnaghHegde** [comment]
@BrainBuzzer I am interested in this. Can you brief me a little. I will work on this.

**Comment by Abhi2424shek** [comment]
Sure @AnaghHedge

**Comment by AnaghHegde** [comment]
@Abhi2424shek Can you please brief me a little about the issue.

**Comment by BrainBuzzer** [comment]
Currently, the problem is that when you go to console rule app, it redirects every time to API link showing JSON Code. That has to be replaced with Asynchronous code. Here are a few links regarding Console Rule app:
https://github.com/fossasia/susi_server/pull/41/files
https://github.com/fossasia/susi_server/issues/29

**Comment by uday96** [comment]
Hey, I was able to implement ajax calls for the test and learn buttons in the console rule app to get the JSON code and display it in a div. But Im not able to test it giving valid field inputs.
Can you brief me about the fields and their inputs? @BrainBuzzer @mariobehling 
![console_ajax](https://cloud.githubusercontent.com/assets/13276887/24967892/2c54134a-1fc9-11e7-9ef5-6571ef85176e.png)

**Comment by Orbiter** [comment]
I don't understand what the original problem is, that shall be solved. 

"The idea is to use AJAX "
...

The idea is, that we don't use AJAX. AJAX is XML+HTTP. We don't use any XML.

You are also addressing concurrency. Concurrency where?

**Comment by BrainBuzzer** [comment]
@Orbiter Even google says that. If we don't have any XML, why do we call it AJAX? It is currently moved to `fetch API` but still let me rephrase that for you.

The idea is to use the Asynchronous code in order to ensure we make the minimum amount of load time and a good UX. Currently, the page redirects every time a query is passed. But the thing is that this can be presented right in front of the user without actually redirecting them.

**Comment by BrainBuzzer** [comment]
@uday96 perhaps #41 might help you a bit.

**Comment by mariobehling** [comment]
This idea could improve the app. However, let's focus on solving bugs and getting everything to work now. We can follow up on other ideas later. Therefore I am closing this now.
